drivebrain_app/DriveBrainApp.cpp

```diff
@@ -0,0 +1,239 @@
+            out_msg->mutable_desired_rpms()->set_rr(out_struct.desired_rpms.RR);
+        }
+
+        out_msg->mutable_torque_limit_nm()->set_fl(::abs(temp_desired_torques.res_torque_lim_nm.FL));
+        out_msg->mutable_torque_limit_nm()->set_fr(::abs(temp_desired_torques.res_torque_lim_nm.FR));
+        out_msg->mutable_torque_limit_nm()->set_rl(::abs(temp_desired_torques.res_torque_lim_nm.RL));
+        out_msg->mutable_torque_limit_nm()->set_rr(::abs(temp_desired_torques.res_torque_lim_nm.RR));
+
+        {
+            std::unique_lock lk(eth_tx_queue_.mtx);
+            eth_tx_queue_.deque.push_back(out_msg);
+            eth_tx_queue_.cv.notify_all();
+        }
+
+        auto end_time = std::chrono::high_resolution_clock::now();
+        auto elapsed = std::chrono::duration_cast<std::chrono::microseconds>(end_time - start_time);
+        
+        if (loop_chrono_time > elapsed) {
+            std::this_thread::sleep_for(loop_chrono_time - elapsed);
+        }
+    }
+}
+
+void DriveBrainApp::startThreads() {
+    db_service_thread_ = std::make_unique<std::thread>([this]() {
+        spdlog::info("started db service thread");
+        try {
+            while (!stop_signal_.load()) {
+                // Run the io_context as long as stop_signal is false
+                db_service_->run_server(); // Run at least one handler, or return immediately if none
+            }
+        } catch (const std::exception& e) {
+            spdlog::error("Error in drivebrain service thread: {}", e.what());
+        }
+    });
+
+    // what we will do here is have a temporary super-loop.
+    // in this thread we will block on having anything in the rx queue, everything by default goes into the foxglove server (TODO)
+    // if we receive the pedals message, we step the controller and get its output to put intot he tx queue
+    
+    io_context_thread_ = std::make_unique<std::thread>([this]() {
+        spdlog::info("Started io context thread");
+        try {
+            io_context_.run();
+        } catch (const std::exception& e) {
+            spdlog::error("Error in io_context: {}", e.what());
+        }
+    });
+
+    process_thread_ = std::make_unique<std::thread>([this]() { processLoop(); });
+}
+
+void DriveBrainApp::run() {
+    startThreads();
+    
+    while (!stop_signal_.load()) {
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
+    
+    stop();
+}
+
+void DriveBrainApp::stop() {
+    stop_signal_.store(true);
+    
+    if (process_thread_ && process_thread_->joinable()) {
+        process_thread_->join();
+    }
+    spdlog::info("joined main process");
+    io_context_.stop();
+    if (io_context_thread_ && io_context_thread_->joinable()) {
+        io_context_thread_->join();
+    }
+    
+    if (db_service_) {
+        db_service_->stop_server();
+    }
+    if (db_service_thread_ && db_service_thread_->joinable()) {
+        db_service_thread_->join();
+    }
+    spdlog::info("joined io context");
+}
```

Comment:
in keeping with the RAII pattern, this can be contained within the destructor of the DriveBrainApp class instead of in a separate function.